src/Components/Components/src/Sections/SectionContent.cs

```diff
@@ -37,11 +44,54 @@ void IComponent.Attach(RenderHandle renderHandle)
     }
 
     Task IComponent.SetParametersAsync(ParameterView parameters)
+    {
+        // We are not using parameters.SetParameterProperties(this)
+        // because IsDefaultContent is internal property and not a parameter
+        SetCustomParameters(parameters);
+
+        if (SectionName is not null && SectionId is not null)
+        {
+            throw new InvalidOperationException($"Both '{nameof(SectionName)}' and '{nameof(SectionId)}' cannot have values.");
+        }
+        else if (SectionName is not null)
+        {
+            _identifier = SectionName;
+        }
+        else if (SectionId is not null)
+        {
+            _identifier = SectionId;
+        }
+        else
+        {
+            throw new InvalidOperationException($"{nameof(SectionContent)} requires a non-null value either for '{nameof(SectionName)}' or '{nameof(SectionId)}'.");
+        }
+
+        if (!object.Equals(_identifier, _registeredIdentifier) || IsDefaultContent != _registeredIsDefaultContent)
+        {
+            if (_registeredIdentifier is not null)
+            {
+                _registry.RemoveProvider(_registeredIdentifier, this);
+            }
+
+            _registry.AddProvider(_identifier, this, IsDefaultContent);
+            _registeredIdentifier = SectionId;
+            _registeredIsDefaultContent = IsDefaultContent;
+        }
+
+        _registry.NotifyContentChanged(_identifier, this);
+
+        return Task.CompletedTask;
+    }
+
+    private void SetCustomParameters(ParameterView parameters)
```

Comment:
Nit:
```suggestion
    private void SetCustomParameters(in ParameterView parameters)
```
(`ParameterView` is a `readonly struct`, so we can use the `in` keyword to pass it as a read-only reference)